core-plugins/src/test/java/co/cask/hydrator/plugin/batch/ETLSnapshotTestRun.java

```diff
@@ -195,6 +259,21 @@ private void testSource(String sourceETLPlugin, String sourceName, Map<String, I
     Assert.assertEquals(expected, actual);
   }
 
+  private List<String> readTextOutput(Location outputLocation) throws IOException {
+    String data = "";
+    for (Location file : outputLocation.list()) {
+      String fileName = file.getName();
+      // text output format doesn't add extension by default, so looking for all part files.
+      if (fileName.startsWith("part")) {
+        String partData = CharStreams.toString(new InputStreamReader(file.getInputStream(), Charsets.UTF_8));
+        data += partData;
+      }
+    }
+    List<String> records = new ArrayList<>();
```

Comment:
its a little better to read through line by line and add to the collection rather than getting the entire string then splitting it